test/integration/logic/resolvable/UnifyRelationConcludableTest.java

```diff
@@ -813,6 +893,329 @@ public void relation_more_players_than_rule_relation_fails_unify() {
+
+        unifiable(conclusions.get(3), "{$p isa driving-hire; $q isa student-driver;}", parents,  new ArrayList<>());
+        unifiable(conclusions.get(3), "{$p isa driving-hire; $q isa driving-hire;}", parents, Lists.newArrayList(1, 2));
+        unifiable(conclusions.get(3), "{$p isa driving-hire; $q isa organisation;}", parents, Lists.newArrayList(1, 2));
+    }
+
+    @Test
+    public void differentIllegalRelationVariants() {
+        String student = Bytes.bytesToHexString(instanceOf("student").getIID());
+        String organisation = Bytes.bytesToHexString(instanceOf("organisation").getIID());
+        List<String> parents = Lists.newArrayList(
+                String.format("{(taxi: $x, night-shift-driver: $y); $x iid %s;}", organisation),
+                String.format("{(taxi: $x, night-shift-driver: $y); $x iid %s;}", student),
+                String.format("{(employer: $x, night-shift-driver: $y); $y iid %s;}", organisation),
+                String.format("{(employer: $x, night-shift-driver: $y); $y iid %s;}", student)
+        );
+
+        List<String> conclusions = Lists.newArrayList(
+                "(employer: $p, employee: $q) isa employment",
+                "(part-time-employer: $p, part-time-employee: $q) isa part-time-employment",
+                "(taxi: $p, day-shift-driver: $q) isa part-time-driving"
+        );
+        conclusions.forEach(conclusion -> unifiable(conclusion,"{$p isa organisation; $q isa person;}", parents, new ArrayList<>()));
+    }
+
+    private ResourceIterator<Unifier> unifiers(String parent, String ruleConclusion, String rulePremises){
+        Conjunction parentConjunction = resolvedConjunction(parent);
+        Concludable.Relation queryConcludable = Concludable.create(parentConjunction).stream()
+                .filter(Concludable::isRelation)
+                .map(Concludable::asRelation)
+                .findFirst().orElse(null);
+        Rule rule = createRule("test-rule", rulePremises, ruleConclusion);
+        return queryConcludable.unify(rule.conclusion(), conceptMgr);
+    }
+
+    private void nonExistentUnifier(String parent, String ruleConclusion, String rulePremises){
+        assertFalse(unifiers(parent, ruleConclusion, rulePremises).hasNext());
+    }
+
+    private Unifier uniqueUnifier(String parent, String ruleConclusion, String rulePremises){
+        List<Unifier> unifiers = unifiers(parent, ruleConclusion, rulePremises).toList();
+        assertEquals(1, unifiers.size());
+        return unifiers.iterator().next();
+    }
+
+    private void unifiable(String ruleConclusion, String rulePremises, List<String> parents, List<Integer> unifiableParents){
+        for (int parentIndex = 0; parentIndex < parents.size() ; parentIndex++) {
+            String parent = parents.get(parentIndex);
+            assertEquals(
+                    String.format("Unexpected unification outcome at index [%s]:\nconjunction: %s\nconclusion: %s\npremises: %s\n",
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.